### PR TITLE
[AMORO-2912] log file paths for which have different file ids when compaction

### LINKED ISSUE #2912: [Improvement]: Log the path of files for different equality file ids when compaction
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, we just throw an exception if there are any files with different equality file ids when compaction, but cant find the files easily.
```
java.lang.IllegalArgumentException: Equality delete files have different delete fields at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:145) at 
```

### How should we improve?

we can log the path for the files which have different file ids.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klion26**
## Why are the changes needed?

Close #2912 .

## Brief change log

- Log more info(file paths) if there are files contain different equality file ids when compaction

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by klion26** [comment]
@zhongqishang @zhoujinsong could you please have a look at this when you're free.

File: amoro-core/src/main/java/org/apache/amoro/io/reader/CombinedDeleteFilter.java [hunk 1]
```diff
@@ -126,11 +126,23 @@ protected CombinedDeleteFilter(
...
+              firstDeleteFilePath = delete.path().toString();
             } else {
-              Preconditions.checkArgument(
-                  deleteIds.equals(
-                      ImmutableSet.copyOf(ContentFiles.asDeleteFile(delete).equalityFieldIds())),
-                  "Equality delete files have different delete fields");
+              Set<Integer> currentDeleteIds =
+                  ImmutableSet.copyOf(ContentFiles.asDeleteFile(delete).equalityFieldIds());
+              if (!deleteIds.equals(currentDeleteIds)) {
+                LOG.warn(
```
Comment (zhoujinsong) [inline amoro-core/src/main/java/org/apache/amoro/io/reader/CombinedDeleteFilter.java:134]: Logging a separate warning message does not seem to be a good practice. 
How about adding this information to the exception's error message, then we can find it in the exception stack information and the Dashboard UI.

**Review by zhoujinsong — COMMENTED** [review]
@klion26 Thanks for the contribution!
I left a small suggestion, PTAL.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks for the contribution! @klion26

**Review by zhongqishang — APPROVED** [review]
LGTM.

**Comment by klion26** [comment]
@zhoujinsong @zhongqishang thanks for the review and merging!

**Pull request merged**